modules/space/rendering/renderabletravelspeed.cpp

```diff
@@ -0,0 +1,335 @@
+    };
+
+    struct [[codegen::Dictionary(RenderableLightTravel)]] Parameters {
+        // [[codegen::verbatim(TargetInfo.description)]]
+        std::string target;
+
+        // [[codegen::verbatim(SpeedInfo.description)]]
+        std::optional<double> travelSpeed;
+
+        // [[codegen::verbatim(LineColorInfo.description)]]
+        std::optional<glm::vec3> color;
+
+        // [[codegen::verbatim(LineOpacityInfo.description)]]
+        std::optional<float> opacity;
+
+        // [[codegen::verbatim(LineWidthInfo.description)]]
+        std::optional<float> lineWidth;
+        
+        // [[codegen::verbatim(IndicatorLengthInfo.description)]]
+        std::optional<int> indicatorLength;
+        
+        // [[codegen::verbatim(FadeLengthInfo.description)]]
+        std::optional<int> fadeLength;
+    };
+#include "renderabletravelspeed_codegen.cpp"
+} // namespace
+
+namespace openspace {
+
+documentation::Documentation RenderableTravelSpeed::Documentation() {
+    return codegen::doc<Parameters>("base_renderable_renderabletravelspeed");
+}
+
+RenderableTravelSpeed::RenderableTravelSpeed(const ghoul::Dictionary& dictionary)
+    : Renderable(dictionary)
+    , _targetName(TargetInfo)
+    , _travelSpeed(
+        SpeedInfo,
+        distanceconstants::LightSecond,
+        1.0, 
+        distanceconstants::LightSecond
+      )
+    , _indicatorLength(IndicatorLengthInfo, 1, 1, 360)
+    , _fadeLength(FadeLengthInfo, 1, 0, 360)
+    , _lineColor(LineColorInfo, glm::vec3(1.f), glm::vec3(0.f), glm::vec3(1.f))
+    , _opacity(LineOpacityInfo, 1.0, 0.0, 1.0)
+    , _lineWidth(LineWidthInfo, 2.f, 1.f, 20.f)
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+    setRenderBin(RenderBin::Overlay);
+
+    _lineColor = p.color.value_or(_lineColor);
+    addProperty(_lineColor);
+
+    _opacity = p.opacity.value_or(_opacity);
+    addProperty(_opacity);
+
+    _lineWidth = p.lineWidth.value_or(_lineWidth);
+    addProperty(_lineWidth);
+
+    _indicatorLength = p.indicatorLength.value_or(_indicatorLength);
+    addProperty(_indicatorLength);
+
+    _fadeLength = p.fadeLength.value_or(_fadeLength);
+    addProperty(_fadeLength);
+
+    _targetName = p.target;
+    addProperty(_targetName);
+    _targetName.onChange([this]() {
+        if (SceneGraphNode* n = sceneGraphNode(_targetName);  n) {
+            _targetNode = n;
+            _targetPosition = _targetNode->worldPosition();
+            _lightTravelTime = calculateLightTravelTime(
+                _sourcePosition, 
+                _targetPosition
+            );
+            calculateDirectionVector();
+            reinitiateTravel();
+        }
+    });
+
+    _travelSpeed = p.travelSpeed.value_or(_travelSpeed);
+    addProperty(_travelSpeed);
+    _travelSpeed.onChange([this]() {
+        reinitiateTravel();
+    });
+}
+
+void RenderableTravelSpeed::initialize() {
+    _initiationTime = -1;
+    _targetNode = sceneGraphNode(_targetName);
```

Comment:
Would be good to check here if the `_targetNode` is a `nullptr` and throw and exception if it is